front-end/src/screens/MenuBar/index.tsx

```diff
@@ -41,7 +43,11 @@ const MenuBar = ({ className } : Props): JSX.Element => {
 			<Menu.Menu position="right">
 				{username
 					? <>
-						<Menu.Item><Icon name='user circle' inverted /> {username}</Menu.Item>
+						<Dropdown text={username} icon={icon} item={true}>
+							<Dropdown.Menu>
+								<Menu.Item as={Link} to="/settings">Settings</Menu.Item>
+							</Dropdown.Menu>
+						</Dropdown>
```

Comment:
Let's put the logout also in the dropdown.

Reply:
> Found a bug for emails on Signup, the email isn't set correctly. How to reproduce:
> 
> * Signup
> * go to settings
> * either there's nothing in the email field, or there is the address that you used to have in memory (so effectively not the email address of the right user).
> * refreshing the page and things are back on track.
> 
> The JWT is sending the right email so the problem come from the front-end.

Will fix this. User signup is using response and not jwt to set user details context. I did not out email in response as i do not want user type to have email. As it is used in users query also. We do not want email public

Reply:
I have created profile query to get user profile info. This can be done with token only.